packages/core-plugin-api/src/plugin/Plugin.tsx

```diff
@@ -30,9 +30,18 @@ import { AnyApiFactory } from '../apis';
 export class PluginImpl<
   Routes extends AnyRoutes,
   ExternalRoutes extends AnyExternalRoutes,
-> implements BackstagePlugin<Routes, ExternalRoutes>
+  PluginInputOptions extends {},
+> implements BackstagePlugin<Routes, ExternalRoutes, PluginInputOptions>
 {
-  constructor(private readonly config: PluginConfig<Routes, ExternalRoutes>) {}
+  constructor(
+    private readonly config: PluginConfig<
+      Routes,
+      ExternalRoutes,
+      PluginInputOptions
+    >,
+  ) {}
+
+  options: {} | undefined = undefined;
```

Comment:
Let's mark this and `getPluginOptions` as `private`, for clarity

Reply:
Changed for options, for `getPluginsOptions` is not allowed by compiler, as we don't explicitly use this method.
![Screenshot 2022-07-07 at 12 27 37](https://user-images.githubusercontent.com/2265094/177752884-0d30414a-9d41-4c85-9619-abda3159123e.png)